OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -640,6 +629,26 @@ void Joint::updateFromXMLNode(SimTK::Xml::Element& aNode, int versionNumber)
                 childNameElt->setValue(childFrameName + "_offset");
             }
         }
+
+        // Version 30506 replaced Joint's CoordinateSet with a "coordinates"
+        // list property.
+        if (documentVersion <= 30506) {
+            if (aNode.hasElement("CoordinateSet")) {
+                auto coordSetIter = aNode.element_begin("CoordinateSet");
+                auto coordIter = coordSetIter->getRequiredElement("objects")
```

Comment:
If w ehave a WeldJoint, will it have an empty "CoordinateSet"? Maybe check that coordSetIter is not at end before dereferencing?